src/lib_ccx/hardsubx_decoder.c

```diff
@@ -707,12 +703,10 @@ void process_hardsubx_linear_frames_and_normal_subs(struct lib_hardsubx_ctx *har
 			{
 				frame_number++;
 
-				avcodec_decode_video2(hard_ctx->codec_ctx,
-						      hard_ctx->frame,
-						      &got_frame,
-						      &hard_ctx->packet);
-
-				if (got_frame && frame_number % 25 == 0)
+				avcodec_send_packet(hard_ctx->codec_ctx, &hard_ctx->packet);
+				if (avcodec_receive_frame(hard_ctx->codec_ctx,
+							  hard_ctx->frame) == 0 &&
+				    frame_number % 25 == 0)
```

Comment:
Where is this 25 coming from? Not all videos are 25 FPS.

Reply:
I just transformed the old code to this, previously it was `if (got_frame && frame_number % 25 == 0)`